src/open_samus_returns_rando/files/custom/scenario.lua

```diff
@@ -142,4 +143,14 @@ end
 
 function Scenario.UpdateRoomName(new_subarea)
   Game.AddSF(0.0, "RoomNameGui.Update", "s", new_subarea)
+end
+
+function Scenario.RandoOnElevatorUse(from_actor, _ARG_1_, _ARG_2_)
+  local destination = ElevatorDestinations[Scenario.CurrentScenarioID][from_actor.sName]
+  if Scenario.CurrentScenario == destination.scenario then
```

Comment:
Looking at this again, I'm not sure if this is ever valid, because I don't think `Scenario.CurrentScenario` exists. If it does exist, I don't think it's achieving whar we want. But I think it's just always ignored.

Reply:
if it doesn't exist then nothing in this function would work correctly

Reply:
oh nvm I misread. you meant to use `Scenario.CurrentScenarioID`

Reply:
My issue is what it's trying to achieve. All the subareas are different scenarios, so this would also be the else condition

Reply:
In my testing though, the elevators still seemed to use their vanilla inter vs inter callback, since that is referenced in the actors data itself, so maybe this isn't even necessary to have here.

Reply:
If this is required, yeah, you would need to use `Scenario.CurrentScenarioID`.
We have something similiar with the DNA mapping per area, which just uses the the first 3 characters. Of course this would require one stupid special case between s040 and s050.

Reply:
Oh wait, if you really need it you could actually use the exact same mapping.
You could use `Init.tScenarioMapping[Scenario.CurrentScenarioID]` and compare it with `Init.tScenarioMapping[destination.scenario]`

Reply:
I'm pretty confident it's not needed.

Reply:
I'll just test this again without these lines and see how it functions. If it behaves like it should in vanilla then I'll remove it. The actor might handle it already via the callback to the function in the elevator lua

Reply:
So after testing this more, it doesn't seem to make a difference if you use `Inner` or `Inter` here. Both are functioning the same, so something else might be at play which displays the minimap on the lower screen when an elevator is used. Not having any GUI line there does crash the game eventually, so I'll just use `Inter` as it's the safest option for now.